Vagrantfile

```diff
@@ -0,0 +1,65 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# Vagrantfile API/syntax version. Don't touch unless you know what you're doing!
+VAGRANTFILE_API_VERSION = "2"
+
+VIRTUAL_MACHINES = {
+  :trusty => {
+    :hostname => 'st2-packages-trusty',
+    :box => 'ubuntu/trusty64',
+    :ip => '192.168.16.20',
+  },
+  :xenial => {
+    :hostname => 'st2-packages-xenial',
+    :box => 'ubuntu/xenial64',
+    :ip => '192.168.16.21',
+  },
+  :el7 => {
+    :hostname => 'st2-packages-el7',
+    :box => 'centos/7',
+    :ip => '192.168.16.22',
+  },
+}
+
+Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+  VIRTUAL_MACHINES.each do |name, cfg|
+    config.vm.define name do |vm_config|
+      vm_config.vm.hostname = cfg[:hostname]
+      vm_config.vm.box = cfg[:box]
+
+      # Box Specifications
+      vm_config.vm.provider :virtualbox do |vb|
+        vb.name = "#{cfg[:hostname]}"
+        # NOTE: With 2048MB, system slows due to kswapd. Recommend at least 4096MB.
+        vb.customize ['modifyvm', :id, '--memory', '4096']
+        vb.cpus = 2
+      end
+
+      # Sync folder using NFS
+      vm_config.vm.synced_folder '.', '/vagrant', nfs: true
```

Comment:
Can we comment out this as "optional"?

I think default Vagrant sync mechanism (https://www.vagrantup.com/docs/synced-folders/basic_usage.html) is more expected for "defaults" and user can fall back to another one if it doesn't work.